packages/twenty-front/src/pages/settings/developers/webhooks/components/SettingsDevelopersWebhookDetail.tsx

```diff
@@ -72,30 +96,85 @@ export const SettingsDevelopersWebhooksDetail = () => {
 
   const isAnalyticsV2Enabled = useIsFeatureEnabled('IS_ANALYTICS_V2_ENABLED');
 
-  const fieldTypeOptions = [
-    { value: '*', label: 'All Objects' },
-    ...objectMetadataItems.map((item) => ({
-      value: item.nameSingular,
-      label: item.labelSingular,
-    })),
+  const fieldTypeOptions: SelectOption<string>[] = useMemo(
+    () => [
+      { value: '*', label: 'All Objects', Icon: IconNorthStar },
+      ...objectMetadataItems.map((item) => ({
+        value: item.nameSingular,
+        label: item.labelSingular,
+        Icon: getIcon(item.icon),
+      })),
+    ],
+    [objectMetadataItems, getIcon],
+  );
+
+  const actionOptions: SelectOption<string>[] = [
+    { value: '*', label: 'All Actions', Icon: IconNorthStar },
+    { value: 'create', label: 'Create', Icon: IconPlus },
+    { value: 'update', label: 'Update', Icon: IconRefresh },
+    { value: 'delete', label: 'Delete', Icon: IconTrash },
   ];
 
   const { updateOneRecord } = useUpdateOneRecord<Webhook>({
     objectNameSingular: CoreObjectNameSingular.Webhook,
   });
 
+  const cleanAndFormatOperations = (operations: WebhookOperationType[]) => {
+    return Array.from(
+      new Set(
+        operations
+          .filter((op) => isDefined(op.object) && isDefined(op.action))
+          .map((op) => `${op.object}.${op.action}`),
+      ),
+    );
+  };
+
   const handleSave = async () => {
+    const cleanedOperations = cleanAndFormatOperations(operations);
     setIsDirty(false);
     await updateOneRecord({
       idToUpdate: webhookId,
       updateOneRecordInput: {
-        operation: `${operationObjectSingularName}.${operationAction}`,
+        operations: cleanedOperations,
         description: description,
       },
     });
     navigate(developerPath);
   };
 
+  const updateOperation = (
+    index: number,
+    field: 'object' | 'action',
+    value: string | null,
```

Comment:
Value will always be defined, as there is only string values in select dropdownOptions
```suggestion
    value: string,
```

Reply:
value could be null. 
WEBHOOK_EMPTY_OPERATION is null
```
export const WEBHOOK_EMPTY_OPERATION: WebhookOperationType = {
  object: null,
  action: null,
};
```
type 
```
export type WebhookOperationType = {
  object: string | null;
  action: string | null;
};
```